images/router/haproxy/reload-haproxy

```diff
@@ -12,8 +12,13 @@ if [ -f $pid_file ]; then
   old_pid=$(<$pid_file)
 fi
 
+# Assumes that the docker ip is the first entry in the hosts file.
+# This port is non-configurable via the router cmd, it is HAProxy specific.  It should
+# match the port used in the haproxy-config.template
+peer_name=$(awk 'NR==1 {print $1}' /etc/hosts):1937
```

Comment:
> @pweil- aah, k that is different if its using it as an id so would using something like hostname -s work - should be the pod name, AFAICR.

To be clear why this doesn't work and why injecting the client wouldn't work: I cannot get the host name of the other replicas to generate the peer section with identical entries.  This only identifies the current replica.

I have the service name.  I could get the selector from the service, query the apiserver to get the pods that match the selector, and build endpoints and labels based on that

```
[vagrant@openshift-master hello-sni]$ osc get services
NAME            LABELS                                    SELECTOR          IP(S)            PORT(S)
kubernetes      component=apiserver,provider=kubernetes   <none>            172.30.0.2       443/TCP
kubernetes-ro   component=apiserver,provider=kubernetes   <none>            172.30.0.1       80/TCP
router          router=router                             router=router     172.30.208.190   80/TCP
tls-service     <none>                                    name=tls-server   172.30.31.12     2999/tcp
[vagrant@openshift-master hello-sni]$ osc get pods -l "router=router"
POD              IP          CONTAINER(S)   IMAGE(S)                          HOST                            LABELS                                                      STATUS    CREATED      MESSAGE
router-1-1hau2   10.1.0.18                                                    openshift-minion-1/10.245.2.3   deployment=router-1,deploymentconfig=router,router=router   Running   25 minutes   
                             router         openshift/origin-haproxy-router                                                                                               Running   25 minutes   
router-1-56kxa   10.1.1.14                                                    openshift-minion-2/10.245.2.4   deployment=router-1,deploymentconfig=router,router=router   Running   25 minutes   
                             router         openshift/origin-haproxy-router                                                                                               Running   25 minutes   
[vagrant@openshift-master hello-sni]$ 
```
The host name is available in an env var that the router could use for the -L

```
[vagrant@openshift-minion-1 hello-sni]$ docker exec 36c6 env
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=router-1-1hau2
```